src/zimdump.cpp

```diff
@@ -463,134 +461,164 @@ void ZimDumper::dumpFiles(const std::string& directory, bool symlinkdump)
-                   "  -f title  find article\n"
-                   "  -u url    find article by url\n"
-                   "  -t        sort (and find) articles by title instead of url\n"
-                   "  -l        list articles\n"
-                   "  -L        list articles as table\n"
-                   "  -o idx    locate article by index\n"
-                   "  -x        print extra parameters\n"
-                   "  -n ns     specify namespace (default 'A')\n"
-                   "  -D dir    dump all files into directory\n"
-#ifndef _WIN32
-                   "  -s        Use symlink to dump html redirect. Else create html redirect file."
-#endif
-                   "  -v        verbose (print uncompressed length of articles when -i is set)\n"
-                   "                    (print namespaces with counts with -F)\n"
-                   "  -V        print the software version number\n"
-                   "  -Z        dump index data\n"
-                   "  -C        verify checksum\n"
-                   "\n"
-                   "examples:\n"
-                   "  " << argv[0] << " -F wikipedia.zim\n"
-                   "  " << argv[0] << " -l wikipedia.zim\n"
-                   "  " << argv[0] << " -f Auto -i wikipedia.zim\n"
-                   "  " << argv[0] << " -f Auto -d wikipedia.zim\n"
-                   "  " << argv[0] << " -f Auto -l wikipedia.zim\n"
-                   "  " << argv[0] << " -f Auto -l -i -v wikipedia.zim\n"
-                   "  " << argv[0] << " -o 123159 -l -i wikipedia.zim\n"
-                 << std::flush;
-      return -1;
+        article = app.getArticle(args["--idx"].asLong());
+    } else if(args["--url"])
+    {
+        std::string nspace = "A";
+        if (args["--ns"])
+            nspace = args["--ns"].asString();
+        article = app.getArticleByUrl(nspace + "/" + args["--url"].asString());
     }
+    if (article.good() == false)
+    {
+        std::cerr << "Article not found" << std::endl;
+        return -1;
+    }
+    return app.dumpArticle(article);
+}
 
-    // initalize app
-    ZimDumper app(argv[1], titleSort);
-    app.setVerbose(verbose);
-
-    // global info
-    if (fileinfo)
-      app.printInfo();
+int subcmdList(ZimDumper &app, std::map<std::string, docopt::value> &args)
+{
+    bool idx = false;
+    bool url = false;
+    bool details = args["--details"].asBool();
 
-    // namespace info
-    if (nsinfo.isSet())
-      app.printNsInfo(nsinfo);
+    if (args["--idx"]) idx = true;
+    if (args["--url"]) url = true;
 
-    // locate article
-    if (indexOffset.isSet())
-      app.locateArticle(indexOffset);
-    else if (find.isSet())
-      app.findArticle(ns, find, titleSort);
-    else if (url.isSet())
-      app.findArticleByUrl(std::string(url));
+    if (idx || url)
+    {
+        zim::Article article;
+        if (idx)
+        {
+            article = app.getArticle(args["--idx"].asLong());
+        } else if(url)
+        {
```

Comment:
It is a matter of style, but I prefer the brace after the if :
```
if (idx) {
   ...
} else if (url) {
   ...
} else {
   ...
}
```

As it is done for the `for`, `try`, ...

Reply:
For me it is ok if everything follows the same convention. I tried to uniform for the other code also but I think we should have a separated PR to apply the code formatting rules (we can even have a clang-format file with our rules).